diff --git a/setup.py b/setup.py
index 8664c37..75212f4 100644
--- a/setup.py
+++ b/setup.py
@@ -104,6 +104,17 @@ class CMakeBuild(build_ext):
                                                               self.distribution.get_version())
         if not os.path.exists(self.build_temp):
             os.makedirs(self.build_temp)
+
+        # hack for conda-build: retrieve all env vars starting with CONDA_QULACS_,
+        # and set them to be cmake variables
+        extra_args = []
+        for key, value in os.environ.items():
+            if key.startswith('CONDA_QULACS_'):
+                var = key[13:]
+                extra_args.append('-D' + var + '=' + value)
+        print("extra_args:", extra_args)
+        cmake_args += extra_args
+
         subprocess.check_call(['cmake', ext.sourcedir] + cmake_args, cwd=self.build_temp, env=env)
         subprocess.check_call(['cmake', '--build', '.', '--target', 'python'] + build_args, cwd=self.build_temp)
 
